src/bin/am/updater.rs

```diff
@@ -0,0 +1,205 @@
+const AUTOMETRICS_GITHUB_ORG: &str = "autometrics-dev";
+const AUTOMETRICS_AM_REPO: &str = "am";
+
+#[derive(Parser)]
+pub struct Arguments {
+    /// Whenever to ignore Homebrew checks and forcefully update
+    #[clap(long, short)]
+    force: bool,
+}
+
+pub(crate) async fn handle_command(args: Arguments, mp: MultiProgress) -> Result<()> {
+    let release = latest_release().await?;
+
+    if !update_needed(&release)? {
+        info!("Already on the latest version");
+        return Ok(());
+    }
+
+    let new_tag = release.tag_name;
+
+    if is_homebrew() && !args.force {
+        info!("A new version of `am` is available: {new_tag}");
+        info!("You can update by running `brew upgrade am` (or use `am update --force`)");
+        return Ok(());
+    }
+
+    info!("Updating to {new_tag}");
+
+    let asset_needed = asset_needed()?;
+
+    let assets: Option<(&Asset, &Asset)> = release
+        .assets
+        .iter()
+        .filter(|a| a.name.starts_with(asset_needed))
+        .sorted_by(|a, b| a.name.cmp(&b.name))
+        .collect_tuple();
+
+    if assets.is_none() {
+        error!("Could not find release for your target platform.");
+        return Ok(());
+    }
+
+    // .unwrap is safe because we checked above if its none
+    // because of .sorted_by above (which sorts by name), the .sha256 file will be the second one *guaranteed*
+    let (binary_asset, sha256_asset) = assets.unwrap();
+
+    let executable = env::current_exe()?;
+    let temp_exe = executable
+        .parent()
+        .ok_or_else(|| anyhow!("Parent directory not found"))?
+        .join("am_update.part");
+
+    let file = File::create(&temp_exe)?;
+
+    let calculated_checksum = download_github_release(
+        &file,
+        AUTOMETRICS_GITHUB_ORG,
+        AUTOMETRICS_AM_REPO,
+        new_tag.strip_prefix('v').unwrap_or(&new_tag),
+        &binary_asset.name,
+        &mp,
+    )
+    .await?;
+
+    let checksum_line = CLIENT
+        .get(sha256_asset.browser_download_url.clone())
+        .send()
+        .await?
+        .text()
+        .await?;
+
+    let remote_checksum = checksum_line
+        .split_once(' ')
+        .map(|(checksum, _)| checksum)
+        .unwrap_or(&checksum_line);
+
+    if calculated_checksum != remote_checksum {
+        debug!(
+            %remote_checksum,
+            %calculated_checksum, "Calculated sha256 hash does not match the remote sha256 hash"
+        );
+
+        fs::remove_file(&temp_exe).context("Failed to delete file that failed checksum match")?;
+        drop(temp_exe);
+
+        bail!("Calculated sha256 hash does not match the remote sha256 hash");
+    }
+
+    fs::rename(&temp_exe, executable)
+        .context("Failed to rename new executable into old executable")?;
+
+    info!("Successfully updated to {new_tag}");
+    Ok(())
+}
+
+pub(crate) async fn update_check() {
+    let project_dirs =
+        ProjectDirs::from("", "autometrics", "am").expect("home directory does not exisT?");
```

Comment:
We should not panic in the update check. Rather we should just log it and return.